src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,42 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public int hashCode() {
+        if (width != null || length != null || color != null) {
+            int result = 17;
+            result = 31 * result + width.hashCode();
+            result = 31 * result + length.hashCode();
+            result = 31 * result + color.hashCode();
+            return result;
+        } else {
+            return 0;
+        }
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) obj;
+        if (obj.getClass().equals(Rectangle.class)
+                && (width != null || length != null || color != null)) {
+            return width.equals(rectangle.width)
+                    && length.equals(rectangle.length)
+                    && color.equals(rectangle.color);
+        } else {
+            return width == rectangle.width
+                    && length == rectangle.length
+                    && color == rectangle.color;
```

Comment:
so if getClass() gave different results for these objects, we still have a chance to get true from equals()?